### PR TITLE
Refactoring part 2

### PR DESCRIPTION
**Pull request by grolu**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```feature user

```

### CODE REVIEW

**Review by petersutter — APPROVED** [review]
/lgtm

File: pkg/cmd/root/command.go [hunk 1]
```diff
@@ -131,3 +132,93 @@ func initConfig() {
 	targetProvider.TargetFile = filepath.Join(home, targetFilename)
 	factory.GardenHomeDirectory = home
 }
+
```
Comment (holgerkoser) [inline pkg/cmd/root/command.go:135]: I would propose to do the following:
* `pkg/cmd/root/command.go` -> `pkg/cmd/cmd.go`
* `pkg/cmd/root/command_whitebox_test.go` -> `pkg/cmd/cmd_whitebox_test.go `
* `pkg/cmd/root/root_suite_test.go` -> `pkg/cmd/cmd_suite_test.go`

File: pkg/cmd/ssh/ssh.go [hunk 2]
```diff
@@ -794,3 +801,253 @@ func getNodes(ctx context.Context, c client.Client) ([]corev1.Node, error) {
 
 	return nodeList.Items, nil
 }
+
+// Options is a struct to support ssh command
+type Options struct {
```
Comment (holgerkoser) [inline pkg/cmd/ssh/ssh.go:806]: rename to `SSHOptions`

File: pkg/cmd/ssh/ssh.go [hunk 3]
```diff
@@ -794,3 +801,253 @@ func getNodes(ctx context.Context, c client.Client) ([]corev1.Node, error) {
...
+	WaitTimeout time.Duration
+
+	// KeepBastion will control whether or not gardenctl deletes the created
+	// bastion once it exits. By default it deletes it, but we allow the user to
+	// keep it for debugging purposes.
+	KeepBastion bool
+}
+
+// NewOptions returns initialized Options
+func NewOptions(ioStreams util.IOStreams) *Options {
```
Comment (holgerkoser) [inline pkg/cmd/ssh/ssh.go:848]: rename to `NewSSHOptions`

File: pkg/cmd/version/version.go [hunk 4]
```diff
@@ -75,3 +76,36 @@ func runCommand(opt *Options) error {
 
 	return nil
 }
+
+// Options is a struct to support version command
+type Options struct {
```
Comment (holgerkoser) [inline pkg/cmd/version/version.go:81]: rename to `VersionOptions`

File: pkg/cmd/version/version.go [hunk 5]
```diff
@@ -75,3 +76,36 @@ func runCommand(opt *Options) error {
...
+	base.Options
+
+	// Short indicates if just the version number should be printed
+	Short bool
+	// Output defines the output format of the version information. Either 'yaml' or 'json'
+	Output string
+}
+
+// NewOptions returns initialized Options
+func NewOptions(ioStreams util.IOStreams) *Options {
```
Comment (holgerkoser) [inline pkg/cmd/version/version.go:91]: rename to `NewVersionOptions`

File: pkg/cmd/root/command.go (hunk 1 above)
Comment (grolu) [reply pkg/cmd/root/command.go:135]: done

File: pkg/cmd/ssh/ssh.go (hunk 2 above)
Comment (grolu) [reply pkg/cmd/ssh/ssh.go]: done

File: pkg/cmd/ssh/ssh.go (hunk 3 above)
Comment (grolu) [reply pkg/cmd/ssh/ssh.go]: done

File: pkg/cmd/version/version.go (hunk 4 above)
Comment (grolu) [reply pkg/cmd/version/version.go]: done

File: pkg/cmd/version/version.go (hunk 5 above)
Comment (grolu) [reply pkg/cmd/version/version.go]: done

**Review by holgerkoser — APPROVED** [review]
/lgtm

**Review by petersutter — APPROVED** [review]
/lgtm

**Pull request merged**